source/Calamari.Shared/Integration/Proxies/ProxySettingsInitializer.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+
+namespace Calamari.Integration.Proxies
+{
+    public static class ProxySettingsInitializer
+    {
+        public static ProxySettings GetProxySettingsFromEnvironment()
+        {
+            bool useDefaultProxy;
```

Comment:
I'm finding this part a little bit confusing.  We check if we should `useDefaultProxy`. We then check if we should `useCustomProxy`. The custom proxy setting then takes precedence over the default proxy setting. The order of the checks seems out of sync with the order of precedence which had me scratching me head a bit: why are the checks done in one order but the preference is the reverse order? The case where `TentacleUseDefaultProxy = true` and `TentacleProxyHost != empty` is not in the fixture, so I'm assuming the intended behaviour.

Also: `Boolean` can be `bool` for consistency, and the output variable can be inlined.

Reply:
yup it's a bit confusing, `TentacleProxyHost` takes precedence.

We only parse `TentacleUseDefaultProxy` at the top but don't do anything with it. I've moved it down below the `if (useCustomProxy)` check so it's clearer